third_party/jsign-nodejs/index.js

```diff
@@ -0,0 +1,74 @@
+import {spawn} from 'child_process';
```

Comment:
I was thinking of this structure:
```
- third_party/
  - jsign/
    - METADATA
    - LICENSE
    - package.json
    - jsign.jar (maybe)
```

Then in `package.json` we either fetch jsign.jar on install, or list the pre-downloaded jsign.jar.
Make sure you list the right license in the `package.json` license field. This allows for automated audit of dependencies.

You can use `require.resolve('jsign')` to get the package location and derive the jar path:
https://nodejs.org/api/modules.html#modules_require_resolve_request_options

It's unclear to me whether you need the index.js. I'd like to minimize code we write under `third_party/`. If we do, then just expose the jar location in a constant.

So ideally we would place most of the logic you wrote here under `src/` or `scripts/` (since that's where we've been placing the build scripts). Or it can just live in the code that calls the signing.

Reply:
Actually I have a different thinking regarding this. I was planning to treat it completely as a published npm package (that's why it is called `third_party`), just like [openssl-nodejs](https://www.npmjs.com/package/openssl-nodejs). And actually it doesn't hurt if we choose to publish it.

Regarding downloading the latest `jsign`: as @daniellacosse pointed, it might block the entire build if the URL is down or the author changes the release URL. Another benefit of using a fixed version is "reproducible build", so every time we are confident that it will always provide the exactly same binaries. If we would like to upgrade to newer `jsign` version, we can simply increase the version number in `jsign-nodejs/package.json` and include a newer version.

These are my thoughts, it depends on you which method you would like to adopt.

Reply:
Updated according to our offline conversation.